neqo-client/src/main.rs

```diff
@@ -428,11 +428,17 @@ fn main() -> Res<()> {
 
     let mut args = Args::from_args();
 
+    let mut resumption_test = false;
+
     if args.qns_mode {
         match env::var("TESTCASE") {
             Ok(s) if s == "http3" => {}
-            Ok(s) if s == "handshake" || s == "transfer" => {
+            Ok(s) if s == "handshake" || s == "transfer" || s == "retry" => {
+                args.use_old_http = true;
+            }
+            Ok(s) if s == "zerortt" || s == "resumption" => {
                 args.use_old_http = true;
+                resumption_test = true;
```

Comment:
Add a check here for the number of URLs rather than down below; no point in executing the test without this.

```rust
if resumption_test && args.urls.len < 2 {
    eprintln!("Warning: resumption won't work without >1 URL");
    exit(127); // Not sure about this one, but it seems right.
}
```

You probably still need to issue a warning further down, because there might be fewer URLs for a single host, but this should catch the worst problems.